var/spack/repos/builtin/packages/snakemake/package.py

```diff
@@ -53,3 +54,27 @@ class Snakemake(PythonPackage):
         depends_on("py-google-crc32c", type=("build", "run"))
         depends_on("py-google-api-python-client", type=("build", "run"))
         depends_on("py-google-cloud-storage", type=("build", "run"))
+
+    # These variants are not in PyPI/pip, but they are undocumented dependencies
+    # needed to make certain parts of Snakemake work.
+    variant("ftp", default=False, description="Enable snakemake.remote.FTP")
+    with when("+ftp"):
+        depends_on("py-ftputil")
+
+    variant("s3", default=False, description="Enable snakemake.remote.S3")
+    with when("+s3"):
+        depends_on("py-boto3")
+        depends_on("py-botocore")
+
+    variant("http", default=False, description="Enable snakemake.remote.HTTP")
+    with when("+http"):
+        depends_on("py-requests")
+
+    with when("@7:"):
+        depends_on("python@3.7:")
```

Comment:
```suggestion
        depends_on("python@3.7:", type=("build", "run"))
```
Or just remove this line since Spack only supports 3.7+